TeXmacs/progs/convert/markdown/markdownout.scm

```diff
@@ -0,0 +1,483 @@
+ (with ltx (math->latex x)
+   (if (null? leave-newlines?)
+       (string-replace (serialize-latex (md-math* ltx)) "\n" " ")
+       (serialize-latex (md-math* ltx)))))
+
+(define (md-equation x)
+  ;; HACK
+  (let*  ((s (md-math x #t))
+          (s1 (string-replace s "\\[" "\\\\["))
+          (s2 (string-replace s1 "\\]" "\\\\]"))
+          (lines (string-split s2 #\newline)))
+    (with-global num-line-breaks 1
+      (serialize-markdown
+       `(document ,@lines)))))
+
+(define (md-eqref x)
+  (let* ((label (cadr x))
+         (err-msg (string-append "undefined label: '" label "'"))
+         (label-name (ahash-ref labels label err-msg)))
+    (string-append "(" label-name ")")))
+
+(define (md-label x)
+  (let ((label-name (number->string label-nr))
+        (label (cadr x)))
+    (if (not (ahash-ref labels label))
+        (begin (ahash-set! labels label label-name) "")
+        (begin (string-append "duplicate label: '" label "'")))))
+
+(define (md-reference x)
+  (let* ((label (cadr x))
+         (err-msg (string-append "undefined label: '" label "'"))
+         (label-name (ahash-ref labels label err-msg)))
+    label-name))
+
+(define (md-item? x)
+  (and (list>0? x) (func? x 'concat) (== (cadr x) '(item))))
+
+(define (md-list x)
+  (let* ((c (cond ((== (car x) 'itemize) "* ")
+                  ((== (car x) 'enumerate) "1. ")
+                  ((== (car x) 'enumerate-alpha) "a. ")
+                  (else "* ")))
+         (cs (string-concatenate (make-list (string-length c) " ")))
+         (transform
+          (lambda (a)
+            (if (md-item? a) `(concat ,c ,@(cddr a)) a))))
+    (with doc (cAr x)
+      (with-global num-line-breaks 1
+        (with-global indent (indent-increment cs)
+          (with-global first-indent (indent-decrement (string-length c))
+            (serialize-markdown `(document ,@(map transform (cdr doc))))))))))
+
+(define (md-quotation x)
+  (with-global num-line-breaks 1
+    (with-global indent (indent-increment "> ")
+      (serialize-markdown (cAr x)))))
+
+(define (style-text style)
+ (cond ((== style 'strong) "**")
+       ((== style 'em) "*")
+       ((== style 'tt) (md-encoding->tm-encoding "`"))
+       ((== style 'strike) "~~")
+       (else "")))
+
+(define (md-style x)
+  (with st (style-text (car x))
+    (string-concatenate 
+     `(,st ,@(map serialize-markdown (cdr x)) ,st " "))))
+
+(define (md-cite x)
+  (if (not (hugo-extensions?)) ""
+      (string-concatenate
+       (list-intersperse
+        (map (cut string-append "{{< cite " <> " >}}") (cdr x))
+        ", "))))
+
+(define (md-cite-detail x)
+  (with detail (cAr x)
+      (string-append (md-cite (cDr x)) " (" detail ")")))
+
+(define (md-hlink x)
+  (with payload (cdr x)
+    (string-append "[" (serialize-markdown (car payload)) "]"
+                   "(" (cadr payload) ")")))    
+
+(define (md-figure x)
+  "Hugo {{< figure >}} shortcode"
+  (if (hugo-extensions?)
```

Comment:
I think the other case needs handling. The figure can be converted to an image and some text below it for instance. Also, following the pattern below, one could place the body of the current if into an `md-hugo-figure`.